### DISCUSSION TITLE
Overwrite GitHub origin and local main with local edit-branch

### DISCUSSION THREAD

**Discussion by mesirhc** [community/community#56713] [New to GitHub]
Hi everyone,

I think my newbie self has screwed up big time!

I have a small (private for now) repo here. 
Cloned it to my Mac. 
Made a local "edit" branch on my Mac and started editing that branch.

I added a new _main.c_ file in my edit-branch, and deleted  _some-ugly-name.c_ that previously delivered the main() function, tried a commit to local-main ...and broke something!

I would like my edit-branch to "overwrite" local main-branch AND GitHub origin-main.

I guess I could copy all files locally whilst in edit-branch, overwrite origin-main, clone origin-main to Mac, and it all just works.
Is there a less brutal way?

Sorry that I can't reply to any suggestions for 8 hours or so: it is bedtime here. ChrisM

**Comment by airtower-luna** [comment]
> I would like my edit-branch to "overwrite" local main-branch AND GitHub origin-main.

If you mean that you want to set your `main` to the *exact same state* as `edit`, and *permanently* discard any difference (don't want to scare you, just be very clear about the consequences :warning:): Use [`git reset`](https://git-scm.com/docs/git-reset) (see also [Reset Demystified](https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified)) and while on `main` `git reset` to `edit`. Pick `--soft` or `--hard` depending on if you want to overwrite your working tree files.

After double checking `main` looks like you want it, you can force-push to your repository on GitHub. There's no undo, so really double-check. :warning:

**Reply by mesirhc** [reply]
Thanks!
Running (not walking) to Reset Demystified.

Will let you know how it goes. ChrisM

**Reply by mesirhc** [reply]
"Demystified" is an exaggeration :(
I didn't have the courage to try reset, but will re-read the demystification again and possibly understand it better when I'm not fretting about losing a couple of weeks of code.

Persuaded GitHub repo to match local edit-branch (RTFM helps!).
Local repo was still a mess, so deleted it and cloned from GitHub.
Compiles, runs, works!   ChrisM

**Reply by airtower-luna** [reply]
I'm afraid it *can* get pretty complex. That's why I was careful not to provide the basic command (`git reset edit`, called while on `main`) above, because there are a bunch of corner cases and "it depends" (like the soft/hard reset thing). :sweat_smile:

**Reply by mesirhc** [reply]
Fully understand your approach, and BIG thanks for being conservativel.
It is so easy to trash anything (even everything) with a one-liner that is not fully understood!

So sorry I can't mark your suggestion as _the answer_, because I took a different route.
Thank you for your suggestion. Even tho I couldn't pursue it, it did provoke me to RTFM. ChrisM
